src/hotspot/share/code/codeCache.cpp

```diff
@@ -310,9 +310,22 @@ void CodeCache::initialize_heaps() {
   FLAG_SET_ERGO(ProfiledCodeHeapSize, profiled_size);
   FLAG_SET_ERGO(NonProfiledCodeHeapSize, non_profiled_size);
 
+  const size_t ps = page_size(false, 8);
+  // Print warning if using large pages but not able to use the size given
+  if (UseLargePages && ps < LargePageSizeInBytes) {
+      char msg[256];
+      jio_snprintf(msg, sizeof(msg),
+                   "Failed to reserve large page memory for code cache (" SIZE_FORMAT "%s). "
+                   "Reverting to smaller page size (" SIZE_FORMAT "%s).",
+                   byte_size_in_exact_unit(LargePageSizeInBytes), exact_unit_for_byte_size(LargePageSizeInBytes),
+                   byte_size_in_exact_unit(ps), exact_unit_for_byte_size(ps));
+      log_warning(codecache)("%s", msg);
```

Comment:
Also, no need to print into a temp buffer. log_xxx() accepts var-args, so you can pass the format string + args directly.

Reply:
I've used the buffer since the message is used in 2 different places:
https://github.com/openjdk/jdk/blob/e1b09a9fe267d6aa48f4656652411000a4f4d2ee/src/hotspot/share/code/codeCache.cpp#L317-L323
(not 100% sure why, but I've noticed that it is how warnings are handled in other places in the code)

Reply:
Ah. missed the second warning. You don't need both. log_warning should be enough and mirrors what we usually do in other places. I assume "warning" is a relict from a time before UL.

Reply:
Ok. I've changed it to use only `log_warning` with varargs. Thank you!